### PR TITLE
Feature create batch

### PR DESCRIPTION
**Pull request by JustinLTu**
Added BatchService.java (In-Progress), BatchEntity.java (Complete), BatchRepository.java (Just a JPA Interface, BatchController.java (In-Progress).

Added BatchServiceTest.java (Complete), BatchRepositoryTest.java (In-Progress)

### CODE REVIEW

File: src/main/java/com/revature/batchservice/controller/BatchController.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	 * Accepts a HTTP GET request. Mapped to ProjectURL/id/{pathVariable}
+	 * Returns a BatchEntity as a JSON entity based on the given id.
+	 * @param id An Integer that contains the BatchEntity id. The id is passed as a path variable. 
+	 * @return a BatchEntity which has the same id as the given id. Null if no matching 
+	 * 			id was found. Value is returned as a JSON object.
+	 */
+	@GetMapping("/id/{id}")
+	public BatchEntity getBatchById(@PathVariable("id") Integer id) {
+		log.debug("Inside findBatchById");
+		System.out.println("Inside findBatchById");
```
Comment (jurczakn) [inline src/main/java/com/revature/batchservice/controller/BatchController.java]: Remove sysouts, logs should be sufficient

File: src/main/java/com/revature/batchservice/controller/BatchController.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	
+	/**
+	 * Accepts a HTTP GET request. 
+	 * Returns a List of all BatchEntities on the connected database as a JSON object.
+	 * @return a List<BatchEntity> that contains all BatchEntities in the database.
+	 */
+	@GetMapping()
+	public List<BatchEntity> getAllBatches() {
+		log.debug("Inside getAllBatches");
+		System.out.println("Inside getAllBatches");
```
Comment (jurczakn) [inline src/main/java/com/revature/batchservice/controller/BatchController.java]: sysout

File: src/main/java/com/revature/batchservice/controller/BatchController.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	/**
+	 *  Accepts a HTTP POST request.
+	 *  Attempts to add a BatchEntity to the database. 
+	 *  @param be The BatchEntity to add to the database.
+	 * 
+	 */
+	@PostMapping()
+	public void createBatch(@RequestBody BatchEntity be) {
+		log.debug("Inside createBatch");
+		System.out.println("Inside createBatch");
```
Comment (jurczakn) [inline src/main/java/com/revature/batchservice/controller/BatchController.java]: no

File: src/main/java/com/revature/batchservice/controller/BatchController.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	 * Takes in a BatchEntity and updates any matching BatchEntity in the database. If
+	 * no BatchEntity on the database has a matching id, then the given BatchEntity is
+	 * added to the database.
+	 * 
+	 * @param be The BatchEntity to update.
+	 */
+	@PutMapping()
+	public void updateBatch(@RequestBody BatchEntity be) {
+		log.debug("Inside updateBatch");
+		System.out.println("Inside updateBatch");
```
Comment (jurczakn) [inline src/main/java/com/revature/batchservice/controller/BatchController.java]: no

File: src/main/java/com/revature/batchservice/controller/BatchController.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	/**
+	 * Accepts a HTTP DELETE request.
+	 * Takes in a BatchEntity and attempts to delete it from the database. 
+	 * If the give BatchEntity does not exist in the database, the database will not be changed.
+	 * @param be The BatchEntity to delete from the database.
+	 */
+	@DeleteMapping()
+	public void deleteBatch(@RequestBody BatchEntity be) {
+		log.debug("Inside deleteBatch");
+		System.out.println("Inside deleteBatch");
```
Comment (jurczakn) [inline src/main/java/com/revature/batchservice/controller/BatchController.java]: no

File: src/main/java/com/revature/batchservice/driver/Driver.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ */
+public class Driver {
+	
+	@Autowired
+	private static BatchController bc;
+	
+	
+	public static void main(String[] args) {
+		/*
```
Comment (jurczakn) [inline src/main/java/com/revature/batchservice/driver/Driver.java]: no commented out code

File: src/main/resources/beans.xml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jurczakn) [inline src/main/resources/beans.xml]: not necessary when using spring boot
configure in app.yml

File: src/test/java/com/revature/batchservice/tests/BatchControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
+package com.revature.batchservice.tests;
```
Comment (jurczakn) [inline src/test/java/com/revature/batchservice/tests/BatchControllerTest.java:1]: need more thorough tests like validating dates and the like

**Review by jurczakn — CHANGES_REQUESTED** [review]
Also, you are not ignoring your compiled class files.  Add a git ignore and remove all unnecessary files

File: src/main/java/com/revature/batchservice/controller/BatchController.java (hunk 1 above)
Comment (JustinLTu) [reply src/main/java/com/revature/batchservice/controller/BatchController.java]: Newest commit implements your changes.

File: src/main/java/com/revature/batchservice/controller/BatchController.java (hunk 2 above)
Comment (JustinLTu) [reply src/main/java/com/revature/batchservice/controller/BatchController.java]: Newest commit implements your changes.

File: src/main/java/com/revature/batchservice/controller/BatchController.java (hunk 3 above)
Comment (JustinLTu) [reply src/main/java/com/revature/batchservice/controller/BatchController.java]: Newest commit implements your changes.

File: src/main/java/com/revature/batchservice/controller/BatchController.java (hunk 4 above)
Comment (JustinLTu) [reply src/main/java/com/revature/batchservice/controller/BatchController.java]: Newest commit implements your changes.

File: src/main/java/com/revature/batchservice/controller/BatchController.java (hunk 5 above)
Comment (JustinLTu) [reply src/main/java/com/revature/batchservice/controller/BatchController.java]: Newest commit implements your changes.

File: src/main/java/com/revature/batchservice/driver/Driver.java (hunk 6 above)
Comment (JustinLTu) [reply src/main/java/com/revature/batchservice/driver/Driver.java]: Newest commit implements your changes.

File: src/main/resources/beans.xml (hunk 7 above)
Comment (JustinLTu) [reply src/main/resources/beans.xml]: Newest commit implements your changes.

File: src/test/java/com/revature/batchservice/tests/BatchControllerTest.java (hunk 8 above)
Comment (JustinLTu) [reply src/test/java/com/revature/batchservice/tests/BatchControllerTest.java:1]: Newest commit implements your changes.

**Comment by JustinLTu** [comment]
Bita pushed a new commit which should have all the changes you wanted.

**Review by jurczakn — CHANGES_REQUESTED** [review]
Git ignore never actually added

**Comment by jurczakn** [comment]
Also remove the entire target folder.  It is not necessary.

**Comment by JustinLTu** [comment]
Actually added .gitignore and removed target/ folder.

**Review by jurczakn — APPROVED** [review]

**Pull request merged**